### DISCUSSION TITLE
Looping Feature

### DISCUSSION THREAD

**Discussion by EvanMarie** [zanllp/infinite-image-browsing#296] [Ideas]
It seems possible to implement the possibility of building in a looping feature that a user could select with one or two check boxes.  The requirement would be to make the first and last image the same, in a continuous, smooth manner.   

One way I can imagine implementing this is as you approach the last image, you would bring in the first image in the outer regions as the fill, and just have generation in a masked region at the boundary.

You could offer the option to combine this with with a duplication of the prompts in reverse order, which would double the length, but may make for better continuity in the end.  

It would be worth experimenting with bringing in the same seed as the forward generations for the last few generations.  Maybe provide an input for duplicate_seed_frames or something.

I think this would be a huge enhancement.

I have not looked at the codebase, but I might be able to implement this here if it is put on the roadmap.

**Comment by zanllp** [comment]
I'm sorry but I have several questions as I'm not quite sure I understand your description.

What is the purpose of this feature? Is it meant to be used in a grid view or in a fullscreen preview after clicking on an image?

Is the looping feature meant to be applied to a folder or to a group of images generated with similar parameters?

Also, I'm wondering if the looping is done in the following index order: 0 -> 1 -> 2 -> 3 -> 4 -> 5 -> 4 -> 3 -> 2 -> 1 -> 0 or if there's another sequence involved.

**Comment by EvanMarie** [comment]
I'm so sorry, I submitted this to the wrong repo.  Too many infinites around here!
